File: app-server/src/main/java/com/github/arhor/simple/expense/tracker/service/UserService.java
package com.github.arhor.simple.expense.tracker.service;

import org.springframework.security.core.Authentication;

import com.github.arhor.simple.expense.tracker.model.UserRequestDTO;
import com.github.arhor.simple.expense.tracker.model.UserResponseDTO;

public interface UserService {

    Long determineUserId(Authentication auth);

    UserResponseDTO determineUser(Authentication auth);

    UserResponseDTO createNewUser(UserRequestDTO request);

    void createNewUserIfNecessary(Authentication authentication);
}

File: app-server/src/main/kotlin/com/github/arhor/simple/expense/tracker/service/UserService.kt
package com.github.arhor.simple.expense.tracker.service

import com.github.arhor.simple.expense.tracker.model.UserRequestDTO
import com.github.arhor.simple.expense.tracker.model.UserResponseDTO
import org.springframework.security.core.Authentication
import org.springframework.security.core.userdetails.UserDetailsService

interface UserService : UserDetailsService {

    fun getUserById(userId: Long): UserResponseDTO

    fun createNewUser(request: UserRequestDTO): UserResponseDTO

    fun createInternalUserIfNecessary(auth: Authentication)
}
